src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1181,7 +1181,9 @@ private void CheckAllowedSolutionChanges(SolutionChanges solutionChanges)
         private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
         {
             if (projectChanges.OldProject.CompilationOptions != projectChanges.NewProject.CompilationOptions
-                && !this.CanApplyChange(ApplyChangesKind.ChangeCompilationOptions))
+                && !this.CanApplyChange(ApplyChangesKind.ChangeCompilationOptions)
+                && !this.CanApplyCompilationOptionChange(
+                    projectChanges.OldProject.CompilationOptions, projectChanges.NewProject.CompilationOptions, projectChanges.NewProject))
```

Comment:
would it make more sense to have this just check CanApplyCompilationOptionChange, and have CanApplyCompilationOptionChange by-default, call into CanApplyChange(ApplyChangesKind.ChangeCompilationOptions)?

Reply:
this was discussion in detail above.. I really don't know what to do anymore, I'd prefer just not touching it as I didn't break anything here and didn't intend to change this

Reply:
@jasonmalinowski See my comment above.  Do you think having this check CanApplyCompilationOptionChange would make more sense?  Note: the defualt CanApplyCompilationOptionChange could just defer to CanApplyChange(ApplyChangesKind.ChangeCompilationOptions) to pick up default behavior.  

Thoughts?